common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -162,12 +174,24 @@ class HTTPClient(
             throw IOException(NetworkStrings.HTTP_RESPONSE_PAYLOAD_NULL)
         }
 
+        val verificationStatus = if (shouldSignResponse && nonce != null) {
+            verifyResponse(path, connection, payload, nonce)
+        } else {
+            HTTPResult.VerificationStatus.NOT_VERIFIED
+        }
```

Comment:
Hmm `takeIf` expects a predicate, so it would have to be `takeIf { shouldSignResponse }` or something like that, which doesn't feel as clean to me... Lmk if you do have a preference though!